python/nav/web/portadmin/views.py

```diff
@@ -108,8 +109,12 @@ def search(query):
     """Search for something in portadmin"""
     netbox_filters = [
         Q(sysname__icontains=query),
-        Q(ip=query)
     ]
+    try:
+        IP(query)
```

Comment:
This will not work, simply due to the fact that IPy will happily accept all sorts of broken things as an IP address, whereas PostgreSQL will not. Pure integers, for example. Just try searching for the number `5`.

I suggest using `nav.util.is_valid_ip()` function instead :)

Reply:
Added bonus: The code gets much cleaner this way. Thanks!